src/index.js

```diff
@@ -426,6 +435,19 @@ function getFallbackComponentName(component) {
 	}
 	return name;
 }
+
+function getAttributeNameInHtmlCase(name) {
+	if (CAMEL_ATTRS.test(name)) return name;
+
+	if (DASHED_ATTRS.test(name))
+		return name.replace(CAPITAL_REGEXP, (w) => '-' + w.toLowerCase());
+
+	if (COLON_ATTRS.test(name))
+		return name.replace(CAPITAL_REGEXP, (w) => ':' + w.toLowerCase());
+
+	return name.toLowerCase();
+}
```

Comment:
HTML allows mixed-case attribute names, so we shouldn't default to converting things to lowercase.

```suggestion
function getAttributeNameInHtmlCase(name) {
	if (DASHED_ATTRS.test(name)) {
		return name.replace(CAPITAL_REGEXP, '-$1');
	}
	return name.replace(COLON_ATTRS, '$1:$2');
}
```

Reply:
Updated the regex shorthands.
Updated the function name - it's not specific to HTML. Using it for HTML/RSS/XML/SVG.
Without lowercasing by default I think there is going to have to be a big whitelist regex.